eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java

```diff
@@ -110,6 +106,94 @@ void assemblesTogether(@TempDir final Path temp) throws IOException {
         );
     }
 
+    @Test
+    @ExtendWith(OnlineCondition.class)
+    void assemblesTogetherWithVersions(@TempDir final Path temp) throws Exception {
+        final Map<String, CommitHash> hashes = new CommitHashesMap.Fake();
+        final CommitHash master = hashes.get("master");
+        final CommitHash five = hashes.get("0.28.5");
+        final CommitHash six = hashes.get("0.28.6");
+        final Map<String, Path> result = new FakeMaven(temp)
+            .withProgram(
+                "+package org.eolang\n",
+                "[x] > main",
+                "  seq > @",
+                "    QQ.io.stdout|0.28.5",
+                "      \"Hello five\"",
+                "    QQ.io.stdout|0.28.6",
+                "      \"Hello six\""
+            )
+            .with("withVersions", true)
+            .with(
+                "objectionaries",
+                new ObjsDefault(
+                    new MapEntry<>(master, new OyRemote(master)),
+                    new MapEntry<>(five, new OyRemote(five)),
+                    new MapEntry<>(six, new OyRemote(six))
+                )
+            )
+            .execute(AssembleMojo.class)
+            .result();
+        final String stdout = "**/io/stdout";
+        final String fifth = new OnVersioned(stdout, "17f8929.xmir").toString();
```

Comment:
@maxonfjvipon I've already seen the similar code. It looks like we use `ObjectName` to create paths.   It is looks like we abuse `OnVersioned.toString` logic. Yes, it does what we need, but is it correct usage of that object? is `"17f8929.xmir"` a hash?  What will happen if we decide to change `OnVersioned` logic?
Maybe it's better to introduce one more class or a method?